ts/packages/cli/src/helpers/stores/TypeORM.ts

```diff
@@ -0,0 +1,98 @@
+import {
+  AuroraMysqlStore,
+  AuroraPostgreSQLStore,
+  CockroachStore,
+  MSSQLStore,
+  MariaStore,
+  MySQLStore,
+  PostgreSQLStore,
+  SQLiteStore,
+} from '@configu/node';
+import { InitFunctionParameters, SchemeToInit } from './types';
+
+const STI = async ({ uri, parsedUri, userinfo, queryDict }: InitFunctionParameters) => {
+  const [username, password] = userinfo;
+  const { database, region, secretArn, resourceArn } = queryDict;
+  let host = `${parsedUri.host}${parsedUri.path}`;
+  if (parsedUri.port) {
+    host = host.concat(`:${parsedUri.port}`);
+  }
+
+  if (!username || !password || !host || !database) {
+    throw new Error(`invalid store uri ${uri}`);
+  }
+
+  if (
+    (parsedUri.scheme === AuroraMysqlStore.scheme || parsedUri.scheme === AuroraPostgreSQLStore.scheme) &&
+    (!region || !secretArn || !resourceArn)
+  ) {
+    throw new Error(`invalid store uri ${uri}`);
+  }
+
+  return {
+    uri,
+    store: {
```

Comment:
@rannn505, I feel like there should probably be a way to use typescript to construct a new instance based on a given class generically.